### PR TITLE
[Autoscaler] Cluster sync expand user

### LINKED ISSUE #10702: [autoscaler] cluster_synced_files doesn't support `~`
### What is the problem?

`~/filename` is marked as "nonexistent" despite actually existing on the cluster.
*Ray version and other system information (Python version, TensorFlow version, OS):* master

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
sent 10,834 bytes  received 637 bytes  22,942.00 bytes/sec
total size is 684,320  speedup is 59.66
    /home/ubuntu/.local/lib/python3.7/site-packages/ray/autoscaler/ from /home/ubuntu/.local/lib/python3.7/site-packages/ray/autoscaler/
  [4/6] Processing worker file mounts
synced files:  ['~/my-cluster-synced-file']
Does ~/my-cluster-synced-file exist? False
```

```
cluster_name: SYNC_TEST
min_workers: 2
max_workers: 2
initial_workers: 2
autoscaling_mode: default
target_utilization_fraction: 0.8
idle_timeout_minutes: 60
provider:
  type: aws
  region: us-east-1
  cache_stopped_nodes: false
auth:
  ssh_user: ubuntu
head_node:
  BlockDeviceMappings:
  - DeviceName: /dev/sda1
    Ebs:
      VolumeSize: 350
      DeleteOnTermination: true
      VolumeType: gp2
  ImageId: latest_dlami
  InstanceType: t3.xlarge
worker_nodes:
  BlockDeviceMappings:
  - DeviceName: /dev/sda1
    Ebs:
      VolumeSize: 300
      DeleteOnTermination: true
      VolumeType: gp2
  ImageId: latest_dlami
  InstanceType: t3.xlarge

cluster_synced_files:
- ~/my-cluster-synced-file
file_mounts_sync_continuously: true
initialization_commands: []
setup_commands:
- pip install --user https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
head_setup_commands:
- tmux new-session -d "python3 -m http.server"
- pip install --user  https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
- sleep 5
- curl localhost:8000 -o ~/my-cluster-synced-file
worker_setup_commands: []

```

If we cannot run your script, we cannot fix your issue.

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?
 - closes '~' issue with cluster_sync_mounts (maybe this is done incorrectly?) closes #10702 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]

File: python/ray/autoscaler/updater.py [hunk 1]
```diff
@@ -78,17 +78,24 @@ def __init__(self,
...
         self.initialization_commands = initialization_commands
         self.setup_commands = setup_commands
         self.ray_start_commands = ray_start_commands
         self.node_resources = node_resources
         self.runtime_hash = runtime_hash
         self.file_mounts_contents_hash = file_mounts_contents_hash
-        self.cluster_synced_files = cluster_synced_files
+        cluster_synced_files = cluster_synced_files or []
+        self.cluster_synced_files = [
+            os.path.expanduser(path) for path in cluster_synced_files
```
Comment (mkoh-asapp) [inline python/ray/autoscaler/updater.py:97]: Won't this expand to the homedir of the machine from which ray was launched, which is not necessarily the cluster homedir?

File: python/ray/autoscaler/updater.py [hunk 2]
```diff
@@ -78,17 +78,24 @@ def __init__(self,
...
+        ]
```
Comment (ijrsvt) [inline python/ray/autoscaler/updater.py:98]: Can you add a comment here  this this field is **only set on the `NodeUpdater` on the head node**.

**Review by ijrsvt — APPROVED** [review]
Looks good, but please add a comment explaining why this works.

**Comment by richardliaw** [comment]
Actually now I think about this, (following a comment from @mkoh-asapp) I think this is pretty confusing. This will only work on the head-node, but it will also sync from laptop-to-head. 

What if we just force the user to pass in an absolute path?

**Comment by wuisawesome** [comment]
"The same path on the head node will be copied to the worker node. This behavior is a subset of the file_mounts behavior."

I think there was just some confusion about the semantics of `cluster_synced_files`.

**Comment by richardliaw** [comment]
Ah ok - so laptop to head won't happen. Why not just force an absolute path in cluster validation?

**Comment by mkoh-asapp** [comment]
> "The same path on the head node will be copied to the worker node. This behavior is a subset of the file_mounts behavior."
> 
> I think there was just some confusion about the semantics of `cluster_synced_files`.

Yes this was my mistake, I deleted my earlier comment. It wasn't clear to me that the path in `cluster_synced_files` only gets processed on the head node.

**Pull request merged**